src/lib/dlt_user.c

```diff
@@ -90,8 +90,16 @@
 #   define DLT_LOG_FATAL_RESET_TRAP(LOGLEVEL)
 #endif /* DLT_FATAL_LOG_RESET_ENABLE */
 
+enum InitState {
+    INIT_UNITIALIZED,
+    INIT_IN_PROGRESS,
+    INIT_DONE
+};
+
 static DltUser dlt_user;
-static atomic_bool dlt_user_initialised = false;
+static enum InitState dlt_user_init_state = INIT_UNITIALIZED;
```

Comment:
shouldn't this be "static volatile enum ..."  as you're later using atomic_compare_exchange_strong with it?

Reply:
Yes good point, added it :+1:

Reply:
This needs to be:
static _Atomic enum InitState dlt_user_init_state = INIT_UNITIALIZED;
and also:
static _Atomic int dlt_user_freeing = 0;

otherwise it does not compile with clang anymore.